portal-api/src/v1/users/routes.js

```diff
@@ -154,32 +155,67 @@ module.exports.findById = (req, res, next) => {
 };
 
 module.exports.updateById = (req, res, next) => {
-  findOne(req.params._id, (error, user) => {
-    if (error) {
-      next(error);
-    } else if (user) {
-      const errors = applyUpdateRules(user, req.body);
-      if (errors.length) {
-        res.status(400).json({
-          success: false,
-          errors: {
-            count: errors.length,
-            errorList: combineErrors(errors),
-          },
-        });
+  if (req.user.roles.includes(ADMIN)) {
```

Comment:
Rather than performing this check in the controller, there is a middleware that can perform this check called `validateUserHasAtLeastOneAllowedRole` in `src/v1/roles`. Your `src/v1/routes.js` file would then have a section like
```
authRouter
  .route('/users/:_id')
  .get(validateUserHasAtLeastOneAllowedRole({ allowedRoles: [ADMIN] }), users.findById)
  ....
```
with similar implementations for the `put` and `delete` functions.

Reply:
This wouldn't work because the endpoint is used by non-admins when changing their password.
It could be refactored to separate these use cases though